torch/fft/__init__.py

```diff
@@ -9,18 +9,283 @@
 # connects the torch.fft Python namespace to the torch._C._fft builtins.
 
 fft = _add_docstr(_fft.fft_fft, r"""
-fft(input) -> Tensor
+fft(input, n=None, dim=-1, norm=None) -> Tensor
 
 Computes the one dimensional discrete Fourier transform of :attr:`input`.
 
+Note:
+
+    The Fourier domain representation of any real signal satisfies the
+    Hermitian property: `X[i] = conj(X[-i])`. This function always returns both
+    the positive and negative frequency terms, even though for real inputs the
+    negative frequencies are redundant. :func:`~torch.fft.rfft` returns the more
+    efficient "half-complex" representation where only the positive frequencies
+    are stored explicitly.
+
 Args:
-  input (Tensor): the input tensor
+    input (Tensor): the input tensor
+    n (int, optional): Signal length. If given, the input will either be zero-padded
+        or trimmed to this length before taking the FFT.
+    dim (int, optional): The dimension along which to take the one dimensional FFT.
+    norm (str, optional): Normalization mode. Between the forward and backward
```

Comment:
This is much better and I think it's already reasonably clear, but I'd still like to try and tweak it once more because this documentation assumes the reader intends to compute the fft and then invert it.  What if we structured this as follows:

- there are 3 normalization modes, forward, backward, and ortho
(bullet points)
- forward normalization mode will cause fft to normalize by 1/n
- backward normalization mode will cause fft to apply no normalization
- ortho will cause fft to apply the 1/sqrt(n) normalization
(end of bullet points)
- the default is "backward" normalization (no normalization)
- These normalization modes correspond with the normalization modes for ifft. For example, the "backward" normalization mode causes fft, the forward function, to apply no normalization, but "backward" normalization mode causes ifft, the backward function, to normalize by 1/n. This means that if the inverse function is applied to the output of the fft function and both have the same normalization mode then a normalization factor of 1/n will have been applied.

That's not perfect as written, but the idea is to capture just the docs for fft devoid of context, then add that perspective. Essentially it's reversing the order of what you've written. 

What are your thoughts? Is it worth trying to tweak this a little?

Reply:
I'm a bit reluctant because it doesn't relate to the mode names very well. How about this draft:

```
norm (str, optional): Normalization mode. For the forward transform (:func:`~torch.fft.fft`),
    these correspond to:

    * ``"forward"`` - normalize by ``1/n``
    * ``"backward"`` - no normalization
    * ``"ortho"`` - normalize by ``1/sqrt(n)`` (making the FFT orthonormal)

    Calling the backward transform (:func:`~torch.fft.ifft`) with the same
    normalization mode will apply an overall normalization of ``1/n`` between
    the two transforms. This is required to makes `ifft` the exact inverse.
```

Reply:
Yes your suggestion is succinct, clear, and much better. Do you want to still put something like, "The default is 'backward' (no normalization)." at the end?

Reply:
I've updated all the docs to this and left the default line as it was.